### ISSUE TITLE
Create notification service wrapper

### ISSUE THREAD

**Issue by dneukirchen** [joomla-projects/media-manager-improvement#148]
We decided to not use Joomla.renderMessages and use our own notification system inside the vue application instead while keeping the look and feel from joomla notifications. We need more control over the notifications (i.e. show a cancelable upload progress notification).

The next step is to evaluate available 3dp vue notification plugins and see if any of those fit our needs. Otherwise we need to create a SIMPLE! notification system by ourself. 

Some basic requirements:

* It has a title
* It has an optional message 
* It supports different types of notifications: success, error, info, warning, custom
* It supports html in message container
* It is possible to stack multiple notifications 
* User can close notifications
* Notifications can be sticky
* Notifications can contain action buttons i.e. cancel upload, redo/undo
* Notifications have a onClose callback
* Notification have a configurable lifetime/timeout
* It uses bootstrap 4 alert styles to keep joomla look and feel
* Notifications fade out on closea

This needs to be implemented before we can complete #162 #163 #164   

I talked to @dgt41 about possible template issues with this approach. The plan for j4 is to decouple the notifications from the template, so it should be ok that mm uses its own notification system as long as we keep the look and feel and give 3dp the possibility to override it.

**Comment by kasvith** [comment]
Can you give more details about the issue, so i can also contribute

**Comment by dneukirchen** [comment]
We need a way to send notifications to the user. 

Joomla offers`Joomla.renderMessages = function( messages, selector, keepOld, timeout)`. This method needs to be wrapped into a vue notification plugin, so that each component can use it and send notifications. It should not be applied to all components. 

Would be nice if we simplify the naming and reduce the number of arguments without loosing flexibility.

**Comment by dneukirchen** [comment]
We decided to not use Joomla.renderMessages and use our own notification system inside the vue application instead while keeping the look and feel from joomla notifications. We need more control over the notifications (i.e. show a cancelable upload progress notification).

The next step is to evaluate available 3dp vue notification plugins and see if any of those fit our needs. 

This needs to be implemented before we can complete #162 #163 #164   

I talked to @dgt41 about possible template issues with this approach. The plan for j4 is to decouple the notifications from the template, so it should be ok, if mm uses its own notification system as long as we keep the look and feel and give 3dp the possibility to override notifications.
